### DISCUSSION TITLE
yolo/datasets/detect/brain-tumor/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#22913] [Docs]
# yolo/datasets/detect/brain-tumor/

探索包含MRI/CT图像的脑肿瘤检测数据集。对于训练用于早期诊断和治疗计划的AI模型至关重要。

https://www.aidoczh.com/yolo/datasets/detect/brain-tumor/

**Comment by G-qingqian** [comment]
Can your dataset ensure that the data are independently and identically distributed by patient? That is, there will be no data from the same patient in both the training set and the validation set.

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for starting this discussion about brain tumor detection datasets with YOLO 🚀! This is an automated Ultralytics assistant response 🤖 — an Ultralytics engineer will also review your discussion and assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions about training custom medical or MRI/CT datasets may already be answered.

If this is a 🐛 Bug Report related to using this brain tumor dataset or its integration with YOLO, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it efficiently.

If this is a custom training ❓ Question (for example, about how to structure the brain tumor dataset, annotations, or best practices for medical imaging), please provide as much information as possible, including:
-Example dataset images (MRI/CT samples)
-Training logs and command-line arguments
-A brief description of your task (detection vs. segmentation, tumor classes, etc.)
Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
-For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧  
-For in-depth discussions, check out [Discourse](https://community.ultralytics.com/)  
-Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community  

## Upgrade

Before proceeding, please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify any issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

These environments are ideal for experimenting with specialized datasets like MRI/CT brain tumor images 🧠.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.
